tests/cloudpickle_test.py

```diff
@@ -2335,6 +2335,47 @@ def check_generic(generic, origin, type_value, use_args):
             assert check_generic(C[int], C, int, use_args) == "ok"
             assert worker.run(check_generic, C[int], C, int, use_args) == "ok"
 
+    def test_generic_subclass(self):
+        T = typing.TypeVar('T')
+
+        class Base(typing.Generic[T]):
+            pass
+
+        class DerivedAny(Base):
```

Comment:
Is deriving from Base (which is a user-defined generic type) without specifying the type `T` a realistic `typing` use case? Otherwise, I think we should just keep the `DerivedInt`/`LeafInt` classes in the test, which, from a quick glance seem to be regular typing/subclassing pattern.

Reply:
I'd say that this is a realistic use case. Consider this scenario: a library author wrote a generic base class, but users of this library may not be that proficient with typing (or they chose to not type annotate stuff at all), so they inherit from the generic base class without specifying the type. I've seen this quite a lot actually.

[PEP 484](https://www.python.org/dev/peps/pep-0484/#user-defined-generic-types) also states:
> Subclassing a generic class without specifying type parameters assumes `Any` for each position.

So I think it's important that we support this use case in `cloudpickle` as well.

Reply:
Thanks for the clarification, that's convincing.